From 17eb7fe58015d36106097867482e4ab9669bf9bf Mon Sep 17 00:00:00 2001
From: codeworkx <daniel.hillenbrand@codeworkx.de>
Date: Mon, 1 Oct 2018 18:33:32 +0200
Subject: [PATCH 35/48] FingerprintService: add property to prevent cleanup of
 unused fingerprints

Restores Oreo behaviour.

Usage:
    ro.fingerprint.preserve.unused=true

Change-Id: Id032fae5c6ae70ce57a60c6f5d3dbe0a6cd33258
---
 .../com/android/server/fingerprint/FingerprintService.java   | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/fingerprint/FingerprintService.java b/services/core/java/com/android/server/fingerprint/FingerprintService.java
index 9f9ed33f753..237e058c711 100644
--- a/services/core/java/com/android/server/fingerprint/FingerprintService.java
+++ b/services/core/java/com/android/server/fingerprint/FingerprintService.java
@@ -62,6 +62,7 @@ import android.os.RemoteException;
 import android.os.SELinux;
 import android.os.ServiceManager;
 import android.os.SystemClock;
+import android.os.SystemProperties;
 import android.os.UserHandle;
 import android.os.UserManager;
 import android.security.KeyStore;
@@ -101,7 +102,7 @@ import java.util.concurrent.CopyOnWriteArrayList;
 public class FingerprintService extends SystemService implements IHwBinder.DeathRecipient {
     static final String TAG = "FingerprintService";
     static final boolean DEBUG = true;
-    private static final boolean CLEANUP_UNUSED_FP = true;
+    private static final boolean PRESERVE_UNUSED_FP = false;
     private static final String FP_DATA_DIR = "fpdata";
     private static final int MSG_USER_SWITCHING = 10;
     private static final String ACTION_LOCKOUT_RESET =
@@ -336,7 +337,7 @@ public class FingerprintService extends SystemService implements IHwBinder.Death
      * @param userId
      */
     private void doFingerprintCleanupForUser(int userId) {
-        if (CLEANUP_UNUSED_FP) {
+        if (!SystemProperties.getBoolean("ro.fingerprint.preserve.unused", PRESERVE_UNUSED_FP)) {
             enumerateUser(userId);
         }
     }
-- 
2.17.1

